src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookConsumer.java

```diff
@@ -173,25 +182,88 @@ private boolean isEligibleRefs(RefsChangedWebhookEvent event) {
         return true;
     }
 
+    private void getJobs(RefChangedDetails refChangedDetails, BitbucketWebhookTriggerRequest.Builder requestBuilder) {
+        Jenkins.get().getAllItems(ParameterizedJobMixIn.ParameterizedJob.class)
+                .stream()
+                .map(BitbucketWebhookConsumer::toTriggerDetails)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .filter(triggerDetails -> hasMatchingRepository(refChangedDetails, triggerDetails.getJob()))
+                .peek(triggerDetails -> LOGGER.fine("Triggering " + triggerDetails.getJob().getFullDisplayName()))
+                .forEach(triggerDetails -> triggerDetails.getTrigger().trigger(requestBuilder.build()));
+    }
+
+    private void triggerJob(PullRequestWebhookEvent event, RefChangedDetails refChangedDetails) {
+        try (ACLContext ctx = ACL.as(ACL.SYSTEM)) {
+            BitbucketWebhookTriggerRequest.Builder requestBuilder = BitbucketWebhookTriggerRequest.builder();
+            event.getActor().ifPresent(requestBuilder::actor);
+
+            getJobs(refChangedDetails, requestBuilder);
+            Optional<BitbucketServerConfiguration> server = bitbucketPluginConfiguration.getValidServerList()
+                                                                                        .stream()
+                                                                                        .filter(serverConfig -> refChangedDetails.getRepository()
+                                                                                                                                 .getSelfLink()
+                                                                                                                                 .contains(serverConfig.getBaseUrl()))
+                                                                                        .findFirst();
+            if (server.isPresent()) {
```

Comment:
As I said in my other comment, I think this is not the right spot for this. However, I'm happy for you to fix that in that PR rather than this one.

Reply:
Amanda did this in her other PR, so resolving this conversation